packages/utils/telemetry-utils/src/telemetryEventBatcher.ts

```diff
@@ -0,0 +1,146 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { performance } from "@fluid-internal/client-utils";
+
+import { roundToDecimalPlaces } from "./mathTools.js";
+import type {
+	ITelemetryGenericEventExt,
+	ITelemetryLoggerExt,
+	ITelemetryPerformanceEventExt,
+} from "./telemetryTypes.js";
+/**
+ * Expected type of the custom data passed into the logger.
+ */
+// type Data<TKey extends string> = { readonly [key in TKey]: number };
+
+/**
+ * Expected type of the custom data passed into the logger.
+ */
+export interface IMeasuredCodeResult<TKey extends string> extends ITelemetryPerformanceEventExt {
```

Comment:
@alexvy86 @Josmithr 

I think the `IMeasuredCodeResult` interface should extend the `ITelemetryPerformanceEventExt` as `sendPerformanceEvent(event: ITelemetryPerformanceEventExt)`.

Reply:
Can you elaborate on why this interface needs to extend the telemetry one? Is it because the data is missing some expected properties when we attempt to submit it to the underlying logger? If so, can we just add the missing properties at that point? Or do we not have sufficient context to do that?

Reply:
You are right. 

_Initially_, I was trying to make `events[]: IMeasuredCodeResult` which made me to make `IMeasuredCodeResult` to extend `ITelemetryPerformanceEventExt`.

However, the object that implements `IMeasuredCodeResult` is stored inside the `telemetryEvent: ITelemetryPerformanceEventExt` object when `sendData()` calls `sendPerformanceEvent()`, therefore I think I can just make `events[]` to be of type `ITelemetryPerformanceEventExt`.